doc/sphinx/proof-engine/tactics.rst

```diff
@@ -102,35 +101,53 @@ specified, the default selector is used.
 Bindings
 ~~~~~~~~
 
-Tactics that take a term as an argument may also accept :token:`bindings`
-to instantiate some parameters of the term by name or position.
-The general form of a term with :token:`bindings` is
-:n:`@term__tac with @bindings` where :token:`bindings` can take two different forms:
+Tactics that take a term as an argument may also accept :token:`bindings` to
+specify the values to assign unbound variables in a term.
+Bindings can be given by position or name.  Generally these appear in the form
+:n:`@one_term_with_bindings` or :n:`with @bindings`, depending on the tactic.
 
-  .. insertprodn bindings bindings
+For some tactics, bindings for all free variables
+must be provided, such as for :tacn:`induction`, :tacn:`destruct`, :tacn:`elim`
+and :tacn:`case`.  Other tactics automatically generate some or all
+of the bindings from the conclusion or a hypothesis, such as :tacn:`apply` and
+:tacn:`constructor` and its variants.  In this case, only instances
+for the :term:`dependent premises <dependent premise>` that are not bound in
+the conclusion of the relevant term are required (and permitted).
```

Comment:
Sorry, but this is still not good. This paragraph really was specific to the first case only. Example illustrating this:

```coq
Goal 0 = 0.
Fail apply @eq_refl with 0.
(* Not the right number of missing arguments (expected 0). *)
apply @eq_refl with (x := 0).
(* works, even though the (x := 0) part is useless because Coq can infer it on its own. *)
```

Reply:
OK, I moved it to the ``` :n:`{+ @one_term }` ``` case.